actions/generate-and-upload-changelog/action.yml

```diff
@@ -0,0 +1,31 @@
+name: 'Generate and Upload Changelog'
+description: 'Generate and Upload Changelog'
+inputs:
+  CURRENT_VERSION:
+    description: 'current semantic version'
+    required: true
+  NEW_VERSION:
+    description: 'new semantic version'
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Generate Changelog
+      if: ${{ github.event_name != 'pull_request'
```

Comment:
This raises an interesting question: if we abstract out all our steps to separate actions where should the logic for the conditions for calling said actions take place?
My opinion: All logic for calling actions should live in the WorkFlow that does the calling, thus giving us maximum reuse of the Actions themselves, this also stops us from having to create new Actions for when we need different conditions.

Reply:
Agreed, will remove the rest of the conditional logic that determines when actions should be run